src/auto_schedule/utils.cc

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file auto_schedule/utils.cc
+ * \brief Common utilities.
+ */
+
+#include "utils.h"
+
+namespace tvm {
+namespace auto_schedule {
+
+NullStream& NullStream::Global() {
+  static NullStream stream;
+  return stream;
+}
+
+ThreadPool& ThreadPool::Global() {
+  static ThreadPool* pool = new ThreadPool();
+  static int ct = 0;
+
+  ct = (ct + 1) % ThreadPool::REFRESH_EVERY;
+
+  if (ct == 0) {
+    pool->Abort();
+    delete pool;
+    pool = new ThreadPool();
+  }
+
+  if (pool->NumWorkers() == 0) {
+    pool->Launch(std::thread::hardware_concurrency());
+  }
+
+  return *pool;
+}
+
+void parallel_for(int start, int end, std::function<void(int index)> f, int stride) {
```

Comment:
@tqchen Added a temporary implementation of parallel_for here.

Reply:
Thanks @jcf94 , let me try to elaborate further. To simplify the abstraction, we should:

- Add src/support/parallel_for.h
   - Move the threadpool as a detail of parallel_for.cc, remove thread_pool from utils.h
   - It is unclear whether threadpool is needed to implement parallel for, it is very possible that we can just launch n std::thread(because std::thread is quite lightweight in c++)
- Use parallel_for for all necessary usecases of threadpool.

Also consider remove the stride argument, or make it optional since stride is not used.

Reply:
@tqchen Ok, I understand that(the stride argument has been set to 1 in default in utils.h), and it's fine for me the clean these code.
Just confused about the "does not have to change now" above. :)

Reply:
I would like to leave it as follow-up PRs.

Reply:
Yes, we should just remove thread pool

Reply:
Removed from the current code base. cc @tqchen